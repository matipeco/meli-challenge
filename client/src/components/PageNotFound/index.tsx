import { FunctionComponent } from "react"

export const PageNotFound:FunctionComponent = () =>{
  return(
    <h2>Pagina no encontrada</h2>
  )
}