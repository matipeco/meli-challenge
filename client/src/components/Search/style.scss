.header{
    background-color: var(--yellow);
    display: flex;
    div{
      padding: 1rem 0;
      display: flex;
      align-items: center;
      justify-content: center;
      }

    label{
      width: 100%;
    }
    &__form{
      display: flex;
      width: 100%;
    }
  &__logo{
    margin-right:3rem;
    width: auto;
    height: 100%;
  }
  &__input{
    font-size: var(--font-400);
    text-align: start;
    border: none;
    padding: 0.5rem 1rem;
    width: 100%;
    border-top-left-radius: 3px;
    border-bottom-left-radius: 3px;
  }
  &__button{
    border: none;
    padding: 0 1rem;
    border-top-right-radius: 3px;
    border-bottom-right-radius: 3px;
    &-logo{
      width: 1.2rem;
    }
  }

}