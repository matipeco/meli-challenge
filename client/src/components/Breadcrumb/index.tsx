import { FunctionComponent } from "react";
import './style.scss';

type Props ={
  categories: string[];
}

export const Breadcrumb:FunctionComponent<Props> = ({categories}) => {
  if(categories.length === 0) return null;
  return(
    <ol className="breadcrumb">
     {categories.map((cat)=>{
          return <li key={cat}>
              <a href="#">{cat}</a>
            </li>
        })}
    </ol>
  )
}