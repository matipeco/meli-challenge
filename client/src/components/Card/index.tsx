import { FunctionComponent } from "react"
import { Product } from "../../types"
import './style.scss'

type Props = {
  prod: Product;
}

export const Card:FunctionComponent<Props> = ({prod
})=>{
  return(
    <article className="card__container">
      <h1>{prod.title}</h1>
    </article>
  )
}