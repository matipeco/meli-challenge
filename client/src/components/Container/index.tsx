import { FunctionComponent, PropsWithChildren } from "react"
import "./style.scss"

export const Container:FunctionComponent<PropsWithChildren> = ({children})=>{
  return(
    <div className="container">
      {children}
    </div>
  )
}