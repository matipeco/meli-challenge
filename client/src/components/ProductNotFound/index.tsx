import productNot from '../../assets/productnot.svg'
import './style.scss'

export const ProductNotFound = ()=>{
  return(
      <div className='product-not__div'>
        <article className='product-not__article'>
          <img src={productNot} alt="" width="20" height="20" />
         <div>
            <p>No hay publicaciones que coincidan con tu búsqueda</p>
            <ul>
              <li>Revisá la ortografía de la palabra.</li>
              <li>Utilizá palabras más genéricas o menos palabras.</li>
            </ul>
          </div> 
        </article>
      </div>
  )
}