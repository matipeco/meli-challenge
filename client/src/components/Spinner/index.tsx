import './style.scss';
import spinner from '../../assets/spinner.svg';

export const Spinner = ()=>{
  return(
    <div className='spinner__container'>
      <img src={spinner} alt="cargando" width="50" height="50"/>
    </div>
  )
}