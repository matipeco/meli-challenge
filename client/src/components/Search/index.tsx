import { ChangeEvent, FormEvent, FunctionComponent, useState } from "react"
import "./style.scss"
import lupa from "../../assets/ic_Search.png"
import logo from "../../assets/Logo_ML.png"
import { Container } from "../Container"
import { Link, useNavigate } from 'react-router-dom'

export const Search:FunctionComponent = () =>{
  const navigate = useNavigate();

  const currentPath = window.location.pathname;

  const [input, setInput] = useState<string>("");

  const handleSubmit = (ev:FormEvent<HTMLFormElement>)=>{
    if(input === ""){
      ev.preventDefault();
      return
    }
    ev.preventDefault();
    navigate(`/items?q=${input}`);
  }

  const handleChange = (ev: ChangeEvent<HTMLInputElement>)=>{
    setInput(ev.target.value.toLowerCase())
  }
  return(
      <header className="header">
        <Container>
        {currentPath !=="/" &&
         <Link to={"/"}>
            <img className="header__logo" src={logo} alt="" />
         </Link>
         
         }
        <form action="" className="header__form" onSubmit={handleSubmit}>
          <label htmlFor="input">
            <input id="input" className="header__input" 
            type="text" placeholder="Nunca dejes de buscar" 
            value={input} 
            onChange={handleChange}/>
          </label>
          <button className="header__button">
            <img src={lupa} alt="" className="header__button-logo" />
          </button> 
        </form>
        </Container>
      </header>
  )
} 
