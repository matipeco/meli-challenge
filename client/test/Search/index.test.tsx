import { render, screen } from '@testing-library/react'
import { Search } from '../../src/components/Search'
import { BrowserRouter } from 'react-router-dom'
import userEvent from '@testing-library/user-event'

const user = userEvent.setup();

describe("Componente Search", ()=>{
  test("Se renderiza correctamente", ()=>{
    const {container} = render(
      <Search/>, 
      {wrapper: BrowserRouter}
    )
    expect(container).toMatchSnapshot();
  })

  test("Se puede tipear en el input", async()=>{
    render(
      <Search/>, 
      {wrapper: BrowserRouter}
    )
    const input = screen.getByLabelText("Search");
    
    await user.click(input);
    await user.keyboard("arroz");

    expect(input).toHaveValue("arroz")
  })

  test("Redirige en el submit", async()=>{
    render(
      <Search/>, 
      {wrapper: BrowserRouter}
    )
    
    const input = screen.getByLabelText("Search");
    const button = screen.getByRole("button", {name:"buscar producto"})

    await user.click(input);
    await user.keyboard("arroz");
    await user.click(button);

    expect(window.location.pathname).toBe("/items")
    expect(window.location.search).toBe("?q=arroz")

  })
})