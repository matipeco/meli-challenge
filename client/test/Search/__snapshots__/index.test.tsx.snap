// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Componente Search > Se renderiza correctamente 1`] = `
<div>
  <header
    class="header"
  >
    <div
      class="container"
    >
      <a
        href="/"
      >
        <img
          alt=""
          class="header__logo"
          src="/src/assets/Logo_ML.png"
        />
      </a>
      <form
        action=""
        class="header__form"
      >
        <label
          class="header__label"
          for="input"
        >
          Search
        </label>
        <input
          class="header__input"
          id="input"
          placeholder="Nunca dejes de buscar"
          type="text"
          value=""
        />
        <button
          aria-label="buscar producto"
          class="header__button"
        >
          <img
            alt=""
            class="header__button-logo"
            src="/src/assets/ic_Search.png"
          />
        </button>
      </form>
    </div>
  </header>
</div>
`;
